src/backend/distributed/planner/query_colocation_checker.c

```diff
@@ -347,6 +364,27 @@ CreateAllTargetListForRelation(Oid relationId, List *requiredAttributes)
 }
 
 
+/*
+ * FindNonDroppedColumnAttribute returns a form pg attribute of a non-dropped
+ * column.
+ */
+static Form_pg_attribute
+FindNonDroppedColumnAttribute(TupleDesc tupleDesc, int numberOfAttributes)
+{
+	for (int attrNum = 1; attrNum <= numberOfAttributes; attrNum++)
+	{
+		Form_pg_attribute attributeTuple =
+			TupleDescAttr(tupleDesc, attrNum - 1);
+		if (!attributeTuple->attisdropped)
+		{
+			return attributeTuple;
+		}
+	}
+	ereport(ERROR, (errmsg(
+						"Unexpected state: there should be at least one non-dropped column in relation.")));
```

Comment:
why? :)

Reply:
umm, interesting

Reply:
added a basic test for this case as well